openff/qcsubmit/results/results.py

```diff
@@ -486,7 +486,48 @@ def create_basic_dataset(
         Returns:
             The created basic dataset.
         """
-        raise NotImplementedError()
+
+        records = self.to_records()
+
+        dataset = BasicDataset(
+            dataset_name=dataset_name,
+            description=description,
+            dataset_tagline=tagline,
+            driver=driver,
+            metadata={} if metadata is None else metadata,
```

Comment:
Should we be using the same metadata as the original input dataset or be making a new one? I guess this is further complicated if we are combining multiple datasets together.

Reply:
Yeah I wasn't sure about this one either because of the fact that the records may have come from different datasets to begin with. 

I think making the user re-populate this is probably the safest way to go as then they'll need to explicitly think through what they are about to do, rather than something surprising happening by default!